src/metabase/api/collection.clj

```diff
@@ -296,16 +298,22 @@
     (not (zero? v))
     v))
 
-(defn- has-required-parameters? [row]
-  (if-let [template-tags (-> row :dataset_query (json/parse-string keyword) :native :template-tags)]
-    (every? #(or (not (:required %)) (:default %)) (vals template-tags))
-    true))
+(defn- fully-parametrized? [row]
```

Comment:
nit: up to you, but I think this function is complex enough now to deserve a docstring